core-jdk8/src/main/java/org/mapstruct/Mapping.java

```diff
@@ -154,4 +154,15 @@
      * @return the dependencies of the mapped property
      */
     String[] dependsOn() default { };
+
+    /**
+     * In case the source property is null the provided default {@link String} value is set.
+     * If the designated target
+     * property is not of type {@code String}, the value will be converted by applying a matching conversion method or
+     * built-in conversion.
+     *
+     * The default value is ignored when using either {@link #constant()} or {@link #expression()} for the mapping.
```

Comment:
Instead of ignoring the default value, an error should pop up. Just like when specifiying both `constant` and `expression`. Check the usage of `Message.PROPERTYMAPPING_EXPRESSION_AND_CONSTANT_BOTH_DEFINED` for an example.